src/Interpreters/interpretSubquery.cpp

```diff
@@ -49,7 +43,7 @@ std::shared_ptr<InterpreterSelectWithUnionQuery> interpretSubquery(
     const auto * table = table_expression->as<ASTIdentifier>();
 
     if (!subquery && !table && !function)
-        throw Exception("Table expression is undefined, Method: ExpressionAnalyzer::interpretSubquery." , ErrorCodes::LOGICAL_ERROR);
+        throw Exception("Table expression is not subquery/table/function.", error_code);
```

Comment:
I don't remember, why we can see something other than subquery/table/function here?

Reply:
I guess that because literal is fine for regular `IN`